src/compiler/checker.ts

```diff
@@ -8071,8 +8071,18 @@ namespace ts {
             return expression;
         }
 
+        function updateReferences(node: Identifier): void {
+            if (!isSourceFileADefinitionFile(node)) {

```

Comment:
I would only check those when reporting.. the option check is probably more expensive than setting the property?